app/src/main/java/com/codepath/apps/critterfinder/fragments/SwipeablePetsFragment.java

```diff
@@ -66,43 +71,79 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         View currentView = inflater.inflate(R.layout.fragment_swipeable_pets, container, false);
         ButterKnife.bind(this, currentView);
         loadingProgress = (LinearLayout)getActivity().findViewById(R.id.loadingProgress);
+
+        mCardContainer.setAdapter(mCardAdapter);
+        mCardContainer.setFlingListener(this);
         doPetSearch(mSearchFilter);
         return currentView;
     }
 
+    @Override
+    public void removeFirstObjectInAdapter() {}
+
+    @Override
+    public void onLeftCardExit(Object o) {
+        PetModel pet = (PetModel) o;
+        skipPet(pet);
+        mCardAdapter.remove(pet);
+    }
+
+    @Override
+    public void onRightCardExit(Object o) {
+        PetModel pet = (PetModel) o;
+        likePet(pet);
+        mCardAdapter.remove(pet);
+    }
+
+    @Override
+    public void onAdapterAboutToEmpty(int itemsInAdapter) {
+        if (itemsInAdapter > 0) {
+            petSearch.doLoadMorePets(this);
+        }
+    }
+
+    @Override
+    public void onScroll(float scrollProgressPercent) {
+        View view = mCardContainer.getSelectedView();
+        view.findViewById(R.id.item_swipe_right_indicator).setAlpha(scrollProgressPercent < 0 ? -scrollProgressPercent : 0);
+        view.findViewById(R.id.item_swipe_left_indicator).setAlpha(scrollProgressPercent > 0 ? scrollProgressPercent : 0);
+    }
```

Comment:
this reveals the like / not like images. A scroll event fires when the user drags a card left/or right - scrolling through the list